sway/desktop/output.c

```diff
@@ -915,6 +915,33 @@ static struct sway_container *output_get_active_workspace(
 	return workspace;
 }
 
+bool output_has_opaque_lockscreen(struct sway_output *output,
+		struct sway_seat *seat) {
+	if (!seat->exclusive_client) {
+		return false;
+	}
+
+	struct wlr_layer_surface *wlr_layer_surface;
+	wl_list_for_each(wlr_layer_surface, &server.layer_shell->surfaces, link) {
+		if (wlr_layer_surface->output != output->wlr_output) {
+			continue;
+		}
+		struct wlr_surface *wlr_surface = wlr_layer_surface->surface;
+		if (wlr_surface->resource->client != seat->exclusive_client) {
+			continue;
+		}
+		pixman_box32_t output_box = {
+			.x2 = output->swayc->current.swayc_width,
+			.y2 = output->swayc->current.swayc_height,
+		};
+		if (pixman_region32_contains_rectangle(&wlr_surface->current->opaque,
```

Comment:
To be completely standards-compliant, `wlr_surface->current->opaque` should be clipped to the surface's size. Probably better to do this on the wlroots side, with a different field.